chain-api/src/types/UniswapV3Pool.ts

```diff
@@ -0,0 +1,574 @@
+/*
+ * Copyright (c) Gala Games Inc. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import BigNumber from "bignumber.js";
+import { Exclude, Type } from "class-transformer";
+import { IsNumber, IsString, ValidateNested } from "class-validator";
+import { JSONSchema } from "class-validator-jsonschema";
+
+import {
+  ChainKey,
+  ConflictError,
+  DefaultError,
+  checkTicks,
+  computeSwapStep,
+  feeAmountTickSpacing,
+  flipTick,
+  formatBigNumber,
+  getAmount0Delta,
+  getAmount1Delta,
+  getFeeGrowthInside,
+  liquidity0,
+  liquidity1,
+  nextInitialisedTickWithInSameWord,
+  requirePosititve,
+  sqrtPriceToTick,
+  tickCross,
+  tickSpacingToMaxLiquidityPerTick,
+  tickToSqrtPrice,
+  updatePositions,
+  updateTick
+} from "../utils";
+import { BigNumberProperty } from "../validators";
+import { ChainObject } from "./ChainObject";
+import { TokenClassKey } from "./TokenClass";
+import { Bitmap, PositionData, Positions, TickData, TickDataObj } from "./dexDtos";
+import { StepComputations, SwapState } from "./dexTypes";
+
+JSONSchema({
+  description: "Uniswap V3 pool chain object with the core contract functionality."
+});
+export class Pool extends ChainObject {
+  @Exclude()
+  static INDEX_KEY = "GCUP"; //GalaChain Uniswap Pool
```

Comment:
```suggestion
  static INDEX_KEY = "GCUSP"; //GalaChain Uniswap Pool
```
`GCUP` is alredy being used by our UserProfile objects